amanuensis/resources/admin/admin_associated_user.py

```diff
@@ -29,6 +30,16 @@ def get_codes_for_roles():
         roles = udm.get_associated_user_roles(session)
         return {data.role: data.code for data in roles}
 
+def update_associated_user_user_id(logged_user_id, logged_user_email):
+    with flask.current_app.db.session as session:
+        user = udm.associate_user.get_associated_user(session, logged_user_email)
+        if not user:
+            raise NotFound("Associated User does not exist") 
```

Comment:
Either you need to handle this exception in the blueprint where this function is called, or replace this line with `return`. Let's say a user has no projects. Logs in the first time to the portal and goes to the Data request page. This will throw 404 and likely break the frontend page since the frontend is expecting an empty array [] if there are no projects.

Reply:
good catch, I didn't think of that